src/main/java/io/cdap/plugin/gcp/gcs/actions/SourceDestConfig.java

```diff
@@ -125,9 +136,20 @@ public void validate(FailureCollector collector, Map<String, String> arguments)
     if (!containsMacro(NAME_CMEK_KEY)) {
       validateCmekKey(collector, arguments);
     }
+    if (!containsMacro(READ_TIMEOUT)) {
+      validateReadTimeout(collector);
+    }
     collector.getOrThrowException();
   }
-
+  void validateReadTimeout(FailureCollector collector) {
+    if (readTimeout == null) {
+      return;
+    }
+    if (readTimeout < 0) {
+      collector.addFailure("Please enter zero or a positive value"
+        , null).withConfigProperty(READ_TIMEOUT);
```

Comment:
collector.addFailure("Read Timeout cannot be less than 0", "Please enter 0 or positive value").withConfigProperty(READ_TIMEOUT);

```Please enter 0 or positive value``` is a corrective action.

Reply:
Understood. Done.